opentelemetry-sdk/allowed-external-types.toml

```diff
@@ -0,0 +1,18 @@
+# Copyright The OpenTelemetry Authors
+# SPDX-License-Identifier: Apache-2.0
+# This is used with cargo-check-external-types to reduce the surface area of downstream crates from
+# the public API. Ideally this can have a few exceptions as possible.
+allowed_external_types = [
```

Comment:
do we expect to run CI checks for this? Or this is for manually checking

Reply:
Yes ideally in the future we can have check-external-types running on each crate. API,SDKs, Exporters, etc.

Reply:
I think we should not merge this without these files actually being used in CI...

Reply:
+1 let's run the current setup as CI?